### ISSUE TITLE
dynamodb.read_items key_condition_expression and filter_expression used together throws an error for version 3.5.0

### ISSUE THREAD

**Issue by ameliaregina** [aws/aws-sdk-pandas#2614]
### Describe the bug

On previous versions (3.4.0/3.4.1) the function works fine but upgrading to 3.5.0 causes the following error message.
```
ClientError                               Traceback (most recent call last)
Cell In[3], line 3
      1 import awswrangler as wr
      2 from boto3.dynamodb.conditions import Key, Attr
----> 3 df = wr.dynamodb.read_items(
      4      table_name='MyTable-NotificationLog',
      5      key_condition_expression=(Key('sampleId').eq('ABC123')),
      6      filter_expression=Attr("stage").eq("Ready"),
      7     )

File [/opt/homebrew/Caskroom/miniforge/base/envs/monitoring/lib/python3.12/site-packages/awswrangler/_utils.py:178](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/miniforge/base/envs/monitoring/lib/python3.12/site-packages/awswrangler/_utils.py:178), in validate_kwargs..decorator..inner(*args, **kwargs)
    175 if condition_fn() and len(passed_unsupported_kwargs) > 0:
    176     raise exceptions.InvalidArgument(f"{message} `{', '.join(passed_unsupported_kwargs)}`.")
--> 178 return func(*args, **kwargs)

File [/opt/homebrew/Caskroom/miniforge/base/envs/monitoring/lib/python3.12/site-packages/awswrangler/dynamodb/_read.py:720](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/miniforge/base/envs/monitoring/lib/python3.12/site-packages/awswrangler/dynamodb/_read.py:720), in read_items(table_name, index_name, partition_values, sort_values, filter_expression, key_condition_expression, expression_attribute_names, expression_attribute_values, consistent, columns, allow_full_scan, max_items_evaluated, dtype_backend, as_dataframe, chunked, use_threads, boto3_session, pyarrow_additional_kwargs)
    718 # If kwargs are sufficiently informative, proceed with actual read op
    719 if any((partition_values, key_condition_expression, filter_expression, allow_full_scan, max_items_evaluated)):
--> 720     return _read_items(
    721         table_name=table_name,
    722         as_dataframe=as_dataframe,
    723         arrow_kwargs=arrow_kwargs,
    724         use_threads=use_threads,
...
-> 1009     raise error_class(parsed_response, operation_name)
   1010 else:
   1011     return parsed_response

ClientError: An error occurred (ValidationException) when calling the Query operation: Query condition missed key schema element: sampleId
```

Using key_condition_expression and filter_expression on their own still work fine for v3.5.0 but the output of using key_condition_expression alone looks different than before where the resulting dataframe has JSON looking values (example in the last section).

### How to Reproduce

This code snippet breaks for both version 3.5.0 anf 3.5.1
```
df = wr.dynamodb.read_items(
     table_name='MyTable-NotificationLog',
     key_condition_expression=(Key('sampleId').eq('ABC123')),
     filter_expression=Attr("status").eq("Ready"),
    )
```

### Expected behavior

Return a dataframe with the proper filtered values and no Validation error.

| |stage	|sampleId	|timestamp|
|---|---------|--------|-----------------------------|
|0	|Ready	|ABC123 |2024-01-10T15:27:16.271Z|

### OS

Mac

### Python version

3.12.0

### AWS SDK for pandas version

3.5.0

### Additional context

Running the following code results in different formats in different versions. I am unsure if this is intentional but I suspect that this may be playing a role in resulting the error.
```
df = wr.dynamodb.read_items(
     table_name='MyTable-NotificationLog',
     key_condition_expression=(Key('sampleId').eq('ABC123'))
)
```

version 3.4.0
| |stage	|sampleId	|timestamp|
|---|---------|--------|-----------------------------|
|0	|Ready	|ABC123 |2024-01-10T15:27:16.271Z|
|1	|Completeed	|ABC123 |2024-01-11T14:27:04.225Z|

version 3.5.0
| |stage	|sampleId	|timestamp|
|---|---------|--------|-----------------------------|
|0	|{'S':'Ready'}	|{'S':'ABC123'} |{'S':'2024-01-10T15:27:16.271Z'}|
|1	|{'S':'Completed'}	|{'S':'ABC123'} |{'S':'2024-01-11T14:27:04.225Z'}|

**Comment by kukushking** [comment]
HI @ameliaregina the format issue is resolved in 3.5.1. Looking into the ValidationException issue.

**Comment by kukushking** [comment]
The root cause is that `ExpressionAttributeNames` and `ExpressionAttributeValues` are used to provide names/values for both key and filter conditions ([L695](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/dynamodb/_read.py#L695), [L706](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/dynamodb/_read.py#L706)), and one of them ends up overwriting the other. Working on a fix.
